packages/core/src/components/region/RegionSlider/RegionSlider.tsx

```diff
@@ -81,43 +120,59 @@ function RegionSlider({ cmsData, open }: RegionSliderProps) {
     })
   }
 
-  const { state, setState } = useSearch()
-  const pickupPoints = usePickupPoints()
-
-  const selectedPickupPoint = state.selectedFacets.find(
-    ({ key }) => key === 'pickupPoint'
-  )?.value
-
-  const [pickupPointOption, setPickupPointOption] = useState<string | null>(
-    selectedPickupPoint ?? null
-  )
+  const handlePickupPointOnChange = (e: MouseEvent<HTMLInputElement>) => {
+    if (pickupPointOption === e.currentTarget.value) {
+      return setPickupPointOption(null)
+    }
 
-  const handlePickupPointOnChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setPickupPointOption(e.target.value)
+    setPickupPointOption(e.currentTarget.value)
   }
 
   const handlePickupPointUpdate = () => {
-    const shippingFacet = state.selectedFacets.find(
-      (facet) => facet.key === 'shipping'
+    const shippingFacet = searchState.selectedFacets.find(
+      (facet) => facet.key === SHIPPING_FACET_KEY
     )
 
+    if (validatedSession && isChangingPickupPoint) {
+      sessionStore.set(validatedSession)
+    }
+
     // If shipping is not 'pickup-in-point', we need to toggle it
     const facetsToToggle = []
 
-    if (!shippingFacet || shippingFacet.value !== 'pickup-in-point') {
-      facetsToToggle.push({ key: 'shipping', value: 'pickup-in-point' })
+    if (!shippingFacet || shippingFacet.value !== PICKUP_IN_POINT_FACET_VALUE) {
+      facetsToToggle.push({
+        key: SHIPPING_FACET_KEY,
+        value: PICKUP_IN_POINT_FACET_VALUE,
+      })
     }
 
     // Add/update the pickupPoint facet
     if (pickupPointOption) {
-      facetsToToggle.push({ key: 'pickupPoint', value: pickupPointOption })
+      facetsToToggle.push({
+        key: PICKUP_POINT_FACET_KEY,
+        value: pickupPointOption,
+      })
+      const pickupPointFacet = pickupPoints.find(
+        (pickupPoint) => pickupPoint.id === pickupPointOption
+      )
+
+      if (regionSliderType === 'changePickupPoint') {
+        changePickupPoint(pickupPointFacet)
+        changeGlobalPickupPoint(null)
+      }
+
+      if (regionSliderType === 'globalChangePickupPoint') {
+        changePickupPoint(pickupPointFacet)
+        changeGlobalPickupPoint(pickupPointFacet)
+      }
     }
 
     if (facetsToToggle.length > 0) {
-      setState({
-        ...state,
+      setSearchState({
+        ...searchState,
```

Comment:
This is redundant on new searchState logic. 
You could pass only the changed state props
```suggestion
      setSearchState({
```